doc/src/manual/parallel-computing.md

```diff
@@ -1,12 +1,542 @@
+finish. This is needed for unpredictable or unbalanced workloads, where we want to assign more
+work to processes only when they finish their current tasks.
+
+The [`@spawn`](@ref) macro allows to execute an expression on a remote node, choosen automatically,
+by creating a closure that returns a `Future` as result. 
+To specify the process use [`@spawnat`](@ref)that accepts a process identifier `p` and an expression. 
+You can pass a `Future` value to [wait()](@ref) that will wait until a value becomes available or
+to [fetch()](@ref) that will wait for and return the value of the future.
+
+For example, let's try to discover on which process an expression is located:
+```julia
+julia>addprocs(4)
+4-element Array{Int64,1}:
+[2,3,4,5]
+julia>f = @spawn myid();
+
+julia>G = wait(f);
+
+julia>fetch(G)
+2
+```
+
+To check the previous result we can explicit the process
+
+```julia
+julia>f = @spawnat 3 myid();
+
+julia>G = wait(f);
+
+julia>fetch(G)
+3
+```
+
+[`@async`](@ref) is similar to [`@spawn`](@ref), but only runs tasks on the local process. We
+use it to create a "feeder" task for each process. Each task picks the next index that needs to
+be computed, then waits for its process to finish, then repeats until we run out of indices. Note
+that the feeder tasks do not begin to execute until the main task reaches the end of the [`@sync`](@ref)
+block, at which point it surrenders control and waits for all the local tasks to complete before
+returning from the function. As for v0.7 and beyond, the feeder tasks are able to share state via `nextidx` because
+they all run on the same process. No locking is required, since the threads are scheduled cooperatively
+and not preemptively. This means context switches only occur at well-defined points: in this case,
+when [`remotecall_fetch`](@ref) is called. This is the current state of implementation (dev v0.7) and it may change
+for future Julia versions, as it is intended to make it possible to run up to N `Tasks` on M `Process`, aka
+[M:N Threading](https://en.wikipedia.org/wiki/Thread_(computing)#Models). Then a lock acquiring\releasing 
+model for `nextidx` will be needed, as it is not safe to let multiple processes read-write a resource at
+the same time.
+
+## Channels
+
+The section on [`Task`](@ref)s in [Control Flow](@ref) discussed the execution of multiple functions in
+a co-operative manner. [`Channel`](@ref)s can be quite useful to pass data between running tasks, particularly
+those involving I/O operations.
+
+Examples of operations involving I/O include reading/writing to files, accessing web services,
+executing external programs, etc. In all these cases, overall execution time can be improved if
+other tasks can be run while a file is being read, or while waiting for an external service/program
+to complete.
+
+A channel can be visualized as a pipe, i.e., it has a write socket and a read socket :
```

Comment:
"has one write end and one read end". Using `socket` here may incorrectly suggest that Channel uses system sockets.